contracts/BoundaryManager.sol

```diff
@@ -104,9 +115,12 @@ contract BoundaryManager {
 
     function removeRefundReceivers(address tokenAddress, address[] calldata refundReceivers) public {
         RefundBoundary storage safeRefundBoundary = safeRefundBoundaries[msg.sender][tokenAddress];
+
+        // COMMENT: Can it overflow? i.e. if the left side is less than the right side in the subtraction?
         safeRefundBoundary.allowedRefundReceiversCount = safeRefundBoundary.allowedRefundReceiversCount - uint16(refundReceivers.length);
 
         for (uint16 i = 0; i < refundReceivers.length; i++) {
+            // COMMENT: again, here, what is the reason for reverting here instead of ignoring?
```

Comment:
We could argue that the transaction arguments are invalid (e.g., you're passing a duplicate address). Ignoring is a "non-obvious behaviour" in a sense. Also, if we ignore it, we'd have to take care of the counter